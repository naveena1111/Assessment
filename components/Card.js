import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight ,Alert} from 'react-native';

function Game(props) {
	
		let icon_name = '?';
		
		if(props.is_open){
			icon_name = props.number;
		}
		
		return (
			<View style={styles.card}>
				<TouchableHighlight onPress={props.clickCard} activeOpacity={0.75} underlayColor={"#f1f1f1"}>
				{props.is_open?<View style={styles.cardFront}><Text style={{color:"black",fontSize:20}}>{icon_name}</Text></View>:
				<View style={styles.cardBack}><Text style={{color:"white",fontSize:20}}>{icon_name}</Text></View>}
				</TouchableHighlight>		
			</View>
		);
	

	

}


const styles = StyleSheet.create({
	card: {
		flex: 1,
		alignItems: 'center'
	},
	card_text: {
		fontSize: 50,
		fontWeight: 'bold'
	},
	cardFront:{
		height:160,
		width:120,
		backgroundColor:"white",
		marginBottom:10,
		justifyContent:"center",
		alignItems:"center",
		borderRadius:10,
		borderWidth:2,
		borderColor:"white"
	},
	cardBack:{
		height:160,
		width:120,
		backgroundColor:"#4db8ff",
		marginBottom:10,
		justifyContent:"center",
		alignItems:"center",
		borderRadius:10,
		borderWidth:2,
		borderColor:"white"
	}
});

export default Game;