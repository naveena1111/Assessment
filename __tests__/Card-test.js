import 'react-native';
import React from 'react';
import Card from '../components/Card';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Card />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});