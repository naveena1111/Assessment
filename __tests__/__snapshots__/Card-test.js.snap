// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4db8ff",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 160,
          "justifyContent": "center",
          "marginBottom": 10,
          "width": 120,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        ?
      </Text>
    </View>
  </View>
</View>
`;
